harness/determined/cli/dev.py

```
@@ -61,6 +68,201 @@ def curl(args: Namespace) -> None:
+        param = params_d.get(key)
+        if not param:
+            raise ValueError(f"Unknown argument {key}")
+        value = parse_param_value(param, value)
+        if key in kwargs:
+            raise ValueError(f"Argument {key} specified twice")
+        kwargs[key] = value
+
+    assert len(kwargs) <= len(params), "too many arguments"
+    return kwargs
+
+
+def print_resposne(d: Any) -> None:
+    if d is None:
+        return
+    if hasattr(d, "to_json"):
+        cli.render.print_json(d.to_json())
+    elif inspect.isgenerator(d):
+        for v in d:
+            print_resposne(v)
+    else:
+        print(d)
+
+
+def list_bindings(args: Namespace) -> None:
+    for name, params in get_available_bindings(show_unusable=args.show_unusable).items():
+        print(bindings_sig_str(name, params))
+
+
+@authentication.required
+def call_bindings(args: Namespace) -> None:
+    """
+    support calling some bindings with primitive arguments via the cli
+    """
+    sess = cli.setup_session(args)
+    fn_name: str = args.name
+    fns = get_available_bindings(show_unusable=False)
+    try:
+        fn = getattr(bindings, fn_name)
+    except AttributeError:
+        matches = [n for n in fns.keys() if re.match(f".*{fn_name}.*", n, re.IGNORECASE)]
+        if not matches:
+            raise errors.CliError(f"no such binding: {fn_name}")
+        print(f"found {len(matches)} matches for '{fn_name}': {list(enumerate(matches))}")
+        user_idx = (
+            input(
+                "Which one do you want? enter the index to continue"
+                + f"(default: 0 {matches[0]}): "
+            )
+            or "0"
+        )
+        fn_name = matches[int(user_idx.strip())]
+        fn = getattr(bindings, fn_name)
+
+    params = fns[fn_name]
+    try:
+        kwargs = parse_args_to_kwargs(args.args, params)
+        rv = fn(sess, **kwargs)
+    except Exception as e:  # we could explicitly check TypeError but let's provide more hints
```

I'm not sure it's ever right to catch all exception types. What are you trying to be inclusive to catch?

---

TypeError and (since our annotations aren't always accurate or as strict as they could be ) whatever exception our apis throw on error I was just lazy to look up all of those.